internal/core/command/router.go

```diff
@@ -7,41 +7,41 @@
 package command
 
 import (
-	"net/http"
+	commandController "github.com/edgexfoundry/edgex-go/internal/core/command/controller/http"
+	commonController "github.com/edgexfoundry/edgex-go/internal/pkg/controller/http"
+	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
 
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/container"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/bootstrap/handlers"
 	"github.com/edgexfoundry/go-mod-bootstrap/v3/di"
 	"github.com/edgexfoundry/go-mod-core-contracts/v3/common"
-	"github.com/gorilla/mux"
 
-	commandController "github.com/edgexfoundry/edgex-go/internal/core/command/controller/http"
-	commonController "github.com/edgexfoundry/edgex-go/internal/pkg/controller/http"
-	"github.com/edgexfoundry/edgex-go/internal/pkg/correlation"
+	"github.com/labstack/echo/v4"
 )
 
-func LoadRestRoutes(r *mux.Router, dic *di.Container, serviceName string) {
-	// r.UseEncodedPath() tells the router to match the encoded original path to the routes
-	r.UseEncodedPath()
-
+func LoadRestRoutes(r *echo.Echo, dic *di.Container, serviceName string) {
 	lc := container.LoggingClientFrom(dic.Get)
 	secretProvider := container.SecretProviderExtFrom(dic.Get)
 	authenticationHook := handlers.AutoConfigAuthenticationFunc(secretProvider, lc)
 
 	// Common
 	cc := commonController.NewCommonController(dic, serviceName)
-	r.HandleFunc(common.ApiPingRoute, cc.Ping).Methods(http.MethodGet) // Health check is always unauthenticated
-	r.HandleFunc(common.ApiVersionRoute, authenticationHook(cc.Version)).Methods(http.MethodGet)
-	r.HandleFunc(common.ApiConfigRoute, authenticationHook(cc.Config)).Methods(http.MethodGet)
+	r.GET(common.ApiPingRoute, cc.Ping) // Health check is always unauthenticated
+	r.GET(common.ApiVersionRoute, cc.Version, authenticationHook)
```

Comment:
Shouldn't the authentication check come BEFORE the regular handler?

Reply:
Thanks @bnevis-i for catching this. I've added `authenticationHook` to the group level middleware and would be invoked first before other middlewares.